src/App.tsx

```diff
@@ -40,11 +40,13 @@ function App() {
     { apiConnection: apiChain2, chainName: CHAIN_2, polkadotjsUrl: providers[CHAIN_2].polkadotjsUrl }
   ];
 
-  if (!apiChain1.isApiReady || !apiChain2.isApiReady) {
+  const apiReady = apiChain1.isApiReady && apiChain2.isApiReady;
+
+  if (!apiReady) {
```

Comment:
im not sure but I think you should put this in a use effect and the outcome inside a state in order to conditional render

Reply:
IMHO i didn't see the need to implement that pattern in this root component since it will not be reused across the application. This is is pretty straight forward and i would be avoiding  to add  effect + state + conditional rendering.

Anyway your comment is more than valid and for the rest of the application i have applied the exact pattern you have described by using the hook : [useLoadingApi](https://github.com/paritytech/parity-bridges-ui/blob/master/src/hooks/useLoadingApi.ts). Then the value pull from this hook is pretty used on different effects.

That pattern would imply to add an effect and state that purpose is already served this this code. Anyway please take into account that this pattern is not used like that anymore across the code base.

Reply:
This pattern (useEffect) in react functional components is replacing the lifecycle of class components. So its kind of a "prerequisite" in react in order to allow react handle things (instead of doing it in a custom way).
Have a look here:
https://github.com/paritytech/substrate-connect/blob/a5f5d71e5fd59222f0c57e1e3f3b4e5c4d115586/projects/burnr/src/App.tsx#L42